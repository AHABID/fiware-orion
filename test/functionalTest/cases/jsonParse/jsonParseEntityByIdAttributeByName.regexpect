1: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1H",
    "registrationId": "REGEX([0-9a-f]{24})"
}
2: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 457
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc"
            }
        }
    ]
}
3: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT2H",
    "registrationId": "REGEX([0-9a-f]{24})"
}
4: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
5: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
6: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: REGEX(.*)

{
    "errorCode": {
        "code": "404",
        "reasonPhrase": "No context element found"
    }
}
7: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: REGEX(.*)

8: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: REGEX(.*)

9: ++++++++++++++++++++
