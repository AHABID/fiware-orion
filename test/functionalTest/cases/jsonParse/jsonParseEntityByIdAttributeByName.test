# Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
#
# This file is part of Orion Context Broker.
#
# Orion Context Broker is free software: you can redistribute it and/or
# modify it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
#
# Orion Context Broker is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
# General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
#
# For those usages not covered by this license please contact with
# fermin at tid dot es

# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh

--NAME--
JSON Parse EntityByIdAttributeByName
--SHELL-INIT--
dbInit CB
brokerStart CB

--SHELL--
# Make a registration using conv op /NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01
echo "1: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1110"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "XXX"
    }
  ],
  "duration" : "PT1H",
  "providingApplication" : "http://kz.tid.es/abc"
}'
curlJson ${url} "${payload}"

REG_ID=$(echo "$response" | grep -Po '(?<="registrationId" : ")[^"]*')

# Make a discovery to see that the registration went OK
echo "2: ++++++++++++++++++++"
url="/NGSI9/discoverContextAvailability"
payload='{
    "entities": [
        {
            "type": "",
            "isPattern": "false",
            "id": "ENTITY_ID"
        }
    ]
}'
curlJson ${url} "${payload}"

# Modify the first registration
echo "3: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1111"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "YYY"
    }
  ],
  "duration" : "PT2H",
  "providingApplication" : "http://kz.tid.es/abc2",
  "registrationId" : "'$REG_ID'"
}'
curlJson ${url} "${payload}"

# Now discover that very registration again to see the change
echo "4: ++++++++++++++++++++"
url="/NGSI9/discoverContextAvailability"
payload='{
    "entities": [
        {
            "type": "",
            "isPattern": "false",
            "id": "ENTITY_ID"
        }
    ]
}'
curlJson ${url} "${payload}"

# Now try the GET method
echo "5: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"

# And a GET that fails
echo "6: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID2/attributes"

# DELETE and PUT should give errors
echo "7: ++++++++++++++++++++"
curlJsonNoPayload "/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01" "-X DELETE"

echo "8: ++++++++++++++++++++"
url="/NGSI9/contextEntities/ENTITY_ID/attributes/ATTRIBUTE_01"
payload='{
  "metadatas" : [
    {
      "name" : "ID",
      "type" : "string",
      "value" : "1111"
    },
    {
      "name" : "cm2",
      "type" : "string",
      "value" : "YYY"
    }
  ],
  "duration" : "PT1H",
  "providingApplication" : "http://kz.tid.es/abc"
}'
curlJson ${url} "${payload}" "-X PUT"

echo "9: ++++++++++++++++++++"

--REGEXPECT--
1: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1H",
    "registrationId": "REGEX([0-9a-f]{24})"
}
2: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 457
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc"
            }
        }
    ]
}
3: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT2H",
    "registrationId": "REGEX([0-9a-f]{24})"
}
4: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
5: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false",
                        "name": "ATTRIBUTE_01",
                        "type": ""
                    }
                ],
                "entities": [
                    {
                        "id": "ENTITY_ID",
                        "isPattern": "false",
                        "type": ""
                    }
                ],
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
6: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: REGEX(.*)

{
    "errorCode": {
        "code": "404",
        "reasonPhrase": "No context element found"
    }
}
7: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: REGEX(.*)

8: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: REGEX(.*)

9: ++++++++++++++++++++
--TEARDOWN--
brokerStop CB
