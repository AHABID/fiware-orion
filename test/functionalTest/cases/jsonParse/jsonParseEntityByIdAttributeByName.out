1: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:41 GMT

{
    "duration": "PT1H", 
    "registrationId": "538d8d8dcfe8966eac0c210f"
}
2: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 457
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:41 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "ATTRIBUTE_01", 
                        "type": ""
                    }
                ], 
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc"
            }
        }
    ]
}
3: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:41 GMT

{
    "duration": "PT2H", 
    "registrationId": "538d8d8dcfe8966eac0c210f"
}
4: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:41 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "ATTRIBUTE_01", 
                        "type": ""
                    }
                ], 
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
5: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 458
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:42 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "ATTRIBUTE_01", 
                        "type": ""
                    }
                ], 
                "entities": [
                    {
                        "id": "ENTITY_ID", 
                        "isPattern": "false", 
                        "type": ""
                    }
                ], 
                "providingApplication": "http://kz.tid.es/abc2"
            }
        }
    ]
}
6: ++++++++++++++++++++
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: Tue, 03 Jun 2014 08:55:42 GMT

{
    "errorCode": {
        "code": "404", 
        "reasonPhrase": "No context element found"
    }
}
7: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Tue, 03 Jun 2014 08:55:42 GMT

8: ++++++++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST, GET
Date: Tue, 03 Jun 2014 08:55:42 GMT

9: ++++++++++++++++++++
